tutorials/composio-newsletter-summarizer-agent/composio-groq.ipynb

```diff
@@ -0,0 +1,369 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's get started!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import packages\n",
+    "import dotenv\n",
+    "from composio_langchain import App, ComposioToolSet\n",
+    "from langchain_groq import ChatGroq\n",
+    "from datetime import datetime\n",
+    "\n",
+    "# Load environment variables from .env file\n",
+    "dotenv.load_dotenv()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We are going to be using Composio's Gmail and Tavily tools in this guide. To use these tools, we first need to create integrations with these apps. This can be done using the command line. You'll need a Tavily API key, which you can get [here](https://app.tavily.com/)  - "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To create an integration with Gmail, run the command below - "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "composio add gmail"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similarly, to create an integration with Tavily -"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "composio add tavily"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Make sure you have a Groq API key and set it as an environment variable. You can get your key [here](https://console.groq.com/) for free."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize the model\n",
+    "llm = ChatGroq(model=\"llama3.1-70b-versatile\", stop_sequences=[\"\\n\\n\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Initialize Toolset and get tools"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we initialize Composio's toolset which provides all the necessary tools and actions we need. For this guide, we will get the Gmail and Tavily tools from the toolset. The Gmail tools will allow the agent to perform actions like fetching and sending emails. The tavily tool allows the agent to search the web and gather more information about the topics of the newsletter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize the ComposioToolSet\n",
+    "toolset = ComposioToolSet()\n",
+    "\n",
+    "# Get the tools from the ComposioToolSet\n",
+    "gmail_tools = toolset.get_tools(apps=[App.GMAIL])\n",
```

Comment:
Thanks for the PR!

I am getting an error on this line, is there a step that I missed, or could the demo be a bit clearer?

`KeyError: 'appName'`

Reply:
Hi, thanks for taking a look at this!

I've updated the code and the above error should be gone now.

Reply:
Thanks! Working for me now, will merge. Great addition to the cookbook!